plasma_framework/contracts/src/utils/PosLib.sol

```diff
@@ -0,0 +1,67 @@
+pragma solidity 0.5.11;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @dev UTXO position = (blockNumber * BLOCK_OFFSET + txIndex * TX_OFFSET + outputIndex).
+ * TX position = (blockNumber * BLOCK_OFFSET + txIndex * TX_OFFSET)
+ */
+library PosLib {
+    using SafeMath for uint256;
+
+    struct Position {
+        uint256 blockNum;
+        uint256 txIndex;
+        uint16 outputIndex;
+    }
+
+    uint256 constant internal BLOCK_OFFSET = 1000000000;
+    uint256 constant internal TX_OFFSET = 10000;
+
+    /**
+     * @notice Returns transaction position which is an utxo position of zero index output
+     * @param pos UTXO position of the output
+     * @return Position of a transaction
+     */
+    function toStrictTxPos(Position memory pos)
+        internal
+        pure
+        returns (Position memory)
+    {
+        return Position(pos.blockNum, pos.txIndex, 0);
+    }
+
+    /**
+     * @notice Used for calculating exit priority
+     * @param pos UTXO position for the output
+     * @return Identifier of the transaction
+     */
+    function getTxPostionForExitPriority(Position memory pos)
+        internal
+        pure
+        returns (uint256)
+    {
+        return encode(pos) / TX_OFFSET;
+    }
+
+    /**
+     * @notice Encodes a position
+     * @param pos Position
+     * @return Position encoded as an integer
+     */
+    function encode(Position memory pos) internal pure returns (uint256) {
+        return pos.blockNum.mul(BLOCK_OFFSET).add(pos.txIndex.mul(TX_OFFSET)).add(pos.outputIndex);
```

Comment:
using SafeMath does mitigate some issue, at least the one that blockNum * BLOCK_OFFSET would exceed max range of uint256.

But there are still other concerns, eg. `outputIndex < TX_OFFSET`, `txIndex < BLOCK_OFFSET / TX_OFFSET`.